phpBB/includes/functions_download.php

```diff
@@ -146,22 +148,28 @@ function send_file_to_browser($attachment, $upload_dir, $category)
 	}
 
 	// Now send the File Contents to the Browser
-	$size = @filesize($filename);
+	try
+	{
+		$file_info = $storage->file_info($filename);
+		$size = $file_info->size;
+	}
+	catch (\Exception $e)
+	{
+		$size = 0;
+	}
 
 	/**
 	* Event to alter attachment before it is sent to browser.
 	*
 	* @event core.send_file_to_browser_before
 	* @var	array	attachment	Attachment data
-	* @var	string	upload_dir	Relative path of upload directory
```

Comment:
What do you think about adding `$storage` to this?

Reply:
For what?

Reply:
Maybe yes, but right now i dont find any utility for that.
Also is just the service, they can call it from their listener too.

Reply:
ok, fair enough